src/main/java/it/gov/pagopa/pu/debtpositions/repository/BasePIIRepository.java

```diff
@@ -72,4 +72,16 @@ protected Pair<Long, Optional<P>> retrievePII(E noPii) {
     }
   }
 
+  public void delete(F fullDTO) {
+    Pair<E, P> p = piiMapper.map(fullDTO);
+
+    Pair<Long, Optional<P>> retrievedPII = retrievePII(p.getFirst());
+
+    if (retrievedPII != null) {
+      personalDataService.delete(retrievedPII.getFirst());
+    }
+
+    noPIIRepository.delete(p.getFirst());
+  }
```

Comment:
this now could handle directly the NoPII entity
```suggestion
  public void delete(e noPii) {
    personalDataService.delete(e.getPersonalDataId());
    noPIIRepository.delete(e);
  }
```